react/LocationInput.tsx

```diff
@@ -93,17 +100,29 @@ const LocationInput: React.FC<Props> = ({
   }
 
   const handleInputChange: React.ChangeEventHandler<HTMLInputElement> = evt => {
+    if (invalidPostalCode) {
+      setInvalidPostalCode(false)
+    }
+
     setInputValue(evt.target.value)
   }
 
-  const handleInputBlur = () => {
+  const isPatternInvalid = () =>
+    countryRules?.fields.postalCode?.pattern != null &&
+    inputValue.match(countryRules.fields.postalCode.pattern) === null
```

Comment:
Just realized that this could be a const inside `formatAndValidate`, but that don't seems to matter too much